presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewQueryOptimizer.java

```diff
@@ -261,7 +273,16 @@ protected Node visitIdentifier(Identifier node, Void context)
         if (!materializedViewInfo.getBaseToViewColumnMap().containsKey(node)) {
```

Comment:
I think we should be able insert the alias back again here, if we want to preserve the alias, WDYT?

Reply:
Adding alias back would require tracking of where it was used initially. I am ok with skipping aliases for now. We may need to add the support of it back when we support multiple tables.

Reply:
For the column alias, it's remain untouched, please see ```visitSingleColumn()```.

For the table alias, you are right, this is the place we can add the alias back. But it can be a little tricky, because in this ```visitIdentifier()``` node, whether the column has the table alias prefix is unknown.
For example, a query ```SELECT t.a, b FROM base b```
When we process the ```Identifier``` ```a``` here, we need to add the alias ```t``` back to it and return a ```DereferenceExpression``` instead of an ```Identifier```.
When we process the ```Identifier``` ```b```, no work needs to be done.
But within this method, we don't know for ```Identifier``` ```a```, ```b```, which one has prefix and which doesn't.

Therefore more work needs to be done if we want to preserve the table alias. Maybe it should be handled in ```visitDereferenceExpression``` or ```visitExpression```.

At this point of time, I would prefer just leave it as it is and I'd like to hear more about your thoughts.

Reply:
yeah, thats exactly I was trying to say with 2nd commit. we would need to track where it was used.

Reply:
Lets add a brief comment to describe what needs to be done to preserve aliases.

Reply:
Sounds good!

Reply:
Added the comment within `visitSingleColumn()`, where the alias was removed.

Reply:
The current solution is to normalize the plans by removing all the prefixes. This solution doesn't work when there are multiple tables.

A better solution(which is hard to implement in the current presto framework) is to compare the columns/tables based on their semantic identifiers instead of using their string aliases. For example, tables with the same table handles are the same no matter what their aliases are. The columns in the tables can have an integer id or use its original column name as id. And the id is used for comparison instead of the alias. But this will require significant changes.